diff --git a/cipd b/cipd
index f7f4133a..2cfcde3a 100755
--- a/cipd
+++ b/cipd
@@ -39,6 +39,10 @@ case "${UNAME}" in
       ARCH="${ARCH_MAC_OVERRIDE}"
     fi
     ;;
+  os/390)
+    OS=zos
+    ARCH=s390x
+    ;;
   *)
     >&2 echo "CIPD not supported on ${UNAME}"
     exit 1
@@ -76,8 +80,14 @@ if [ -z $ARCH ]; then
       ARCH=riscv64
       ;;
     *)
-      >&2 echo "UNKNOWN Machine architecture: ${UNAME}"
-      exit 1
+      # uname -m changes with machine type (e.g. 3931 for z16, 8561 for z15)
+      UNAME=`uname -s | tr '[:upper:]' '[:lower:]'`
+      if [[ "$UNAME" == "os/390" ]]; then
+        ARCH=s390x
+      else
+        >&2 echo "UNKNOWN Machine architecture: ${UNAME}"
+        exit 1
+      fi
   esac
 fi
 
