diff --git a/detect_host_arch.py b/detect_host_arch.py
index 7669d9ec..67d1e0cb 100755
--- a/detect_host_arch.py
+++ b/detect_host_arch.py
@@ -28,6 +28,8 @@ def HostArch():
         host_arch = 'mips'
     elif host_arch.startswith('ppc') or host_processor == 'powerpc':
         host_arch = 'ppc'
+    elif platform.system() == 'OS/390':
+        host_arch = 's390x'
     elif host_arch.startswith('s390'):
         host_arch = 's390'
     elif host_arch.startswith('riscv'):
