diff --git a/gclient_utils.py b/gclient_utils.py
index e017f996..d0b3e904 100644
--- a/gclient_utils.py
+++ b/gclient_utils.py
@@ -630,7 +630,7 @@ def CheckCallAndFilter(args,
         # subprocess when filtering its output. This makes the subproc believe
         # it was launched from a terminal, which will preserve ANSI color codes.
         os_type = GetOperatingSystem()
-        if sys.stdout.isatty() and os_type != 'win' and os_type != 'aix':
+        if sys.stdout.isatty() and os_type not in ['win', 'aix', 'zos']:
             pipe_reader, pipe_writer = os.openpty()
         else:
             pipe_reader, pipe_writer = os.pipe()
@@ -791,6 +791,9 @@ def GetOperatingSystem():
     if sys.platform.startswith('aix'):
         return 'aix'
 
+    if sys.platform == 'zos':
+        return 'zos'
+
     try:
         return os.uname().sysname.lower()
     except AttributeError:
